### DISCUSSION TITLE
How to create custom authentication?

### DISCUSSION THREAD

**Discussion by alex-oliveira** [adonisjs/core#2657] [Help] [answered]
I am trying to create a customized "provider / guard" to enable user authentication.

I plan to use JWT and a no-SQL database (DynamoDB or FaunaDB), but I am not finding more detailed information on how to do this.

I would like this implementation to be integrated with the way AdonisJs works.

Could someone get me some reference?

**Comment by guntribam** [comment]
Please vote on this issue. We need docs for this. I'm facing this issue as well
https://trello.com/c/x8SCwZR9/8-guide-on-package-development

**Comment by thetutlage** [comment, marked as answer]
I have written the docs for adding a custom user provider for auth here. https://docs.adonisjs.com/guides/auth/custom-user-provider. Here is an example project showcasing it with Mongoose https://github.com/adonisjs-community/auth-mongoose-provider.

The docs for **custom guard** are yet to come

**Reply by guntribam** [reply]
thank you so much
